client/components/domains/contact-details-form-fields/test/index.js

```diff
@@ -51,159 +61,178 @@ describe( 'ContactDetailsFormFields', () => {
 
 	describe( 'default fields', () => {
 		test( 'should render', () => {
-			const wrapper = shallow( <ContactDetailsFormFields { ...defaultProps } /> );
-
-			expect( wrapper ).toMatchSnapshot();
+			const { container } = render( <ContactDetailsFormFields { ...defaultProps } /> );
+			expect( container.firstChild ).toMatchSnapshot();
 		} );
 
 		test( 'should render fields when contact details contains no values', () => {
 			const newProps = { ...defaultProps, contactDetails: {} };
 
-			const wrapper = shallow( <ContactDetailsFormFields { ...newProps } /> );
+			render( <ContactDetailsFormFields { ...newProps } /> );
 
-			expect( wrapper.find( '[name="first-name"]' ) ).toHaveLength( 1 );
-			expect( wrapper.find( '[name="last-name"]' ) ).toHaveLength( 1 );
-			expect( wrapper.find( '[name="phone"]' ) ).toHaveLength( 1 );
+			expect( screen.queryByRole( 'textbox', { name: /first name/i } ) ).toBeInTheDocument();
```

Comment:
I usually prefer `toBeVisible()` rather than `toBeInTheDocument()` because an element can be in the DOM tree, but be invisible. So `toBeInvisible()` is one step further and better reflects our intention when the element should actually be visible to the user.

That comment applies to the rest of the non-`.not` usages of `toBeInTheDocument()` in this PR.

Reply:
I think that's a great suggestion 👍🏻